### PR TITLE
8.0 Add module l10n_be_invoice_ubl to support the e-fff standard

### PR DESCRIPTION
**Pull request by alexis-via**
Left todo:
- [ ] add more mandatory fields, cf http://www.e-fff.be/files/efff_UBL_Belgique_V110_Final.xls
- [ ] Check if we have to switch to UBL 2.0 instead of 2.1
- [ ] validate the files generated by Odoo ?

### CODE REVIEW

File: l10n_be_invoice_ubl/models/account_invoice.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        parent_node.insert(
+            parent_node.index(version_node_find) + 1, customization_id)
+        profile_id = etree.Element(ns['cbc'] + 'ProfileID')
+        profile_id.text = 'EFFF.BE Odoo %s' % release.version
+        parent_node.insert(
+            parent_node.index(version_node_find) + 2, profile_id)
+        return res
+
+    @api.multi
+    def get_ubl_filename(self, version='2.1'):
```
Comment (houssine78) [inline l10n_be_invoice_ubl/models/account_invoice.py:38]: should it be version 2.0?

File: l10n_be_invoice_ubl/models/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        '''Read from http://www.e-fff.be/FR/doc03.php :
+        The e-fff community recommends to use the following file naming:
+        efff_BE0123456789_AlphaNumericCharactersFreeOfChoice.xml'''
+        filename = 'efff_%s_Invoice_%s' % (
+            self.commercial_partner_id.sanitized_vat,
+            self.number and self.number.replace('/', '-') or 'draft')
+        return filename
```
Comment (houssine78) [inline l10n_be_invoice_ubl/models/account_invoice.py:45]: add +'.xml'

File: l10n_be_invoice_ubl/models/account_invoice.py (hunk 1 above)
Comment (alexis-via) [reply l10n_be_invoice_ubl/models/account_invoice.py:38]: This is the proto of the method that we inherit, so we don't change it in the proto.

File: l10n_be_invoice_ubl/models/account_invoice.py (hunk 2 above)
Comment (alexis-via) [reply l10n_be_invoice_ubl/models/account_invoice.py:45]: Good catch !

**Comment by rousseldenis** [comment]
For information, found Peppol website that defines Pan European specifications (UBL)
http://www.peppol.eu/ressource-library/technical-specifications/post-award/mandatory

**Comment by houssine78** [comment]
+1

**Comment by alexis-via** [comment]
Is e-fff so unpopular in Belgium that nobody's interested in this PR ?

@sbidoul I don't see any field set as "Mandatory" in the specs which is not present in the UBL file generated by Odoo (given that the appropriate info is set in Odoo). Could you tell us which field you see as mandatory in the spec that is not set by Odoo ?

**Comment by sbidoul** [comment]
@alexis-via it could be the intersection of the set of e-fff users and the set of odoo users is empty.

We may change that this year though. Patience.

Regarding the mandatory fields, I don't know yet. That todo was yours :)

**Comment by sbidoul** [comment]
See also https://github.com/odoo/odoo/pull/17278

**Pull request closed without merging**